File: code/57267.py
def ktdoixung(s):
    n=len(s)-1
    i=0
    while i<n and s[i]==s[n]:
        i+=1
        n-=1
    return i>=n
s=input()
n=len(s)
for i in range(n):
    for j in range(i,n):
        p=s[i:j+1]
        if ktdoixung(p): 
            print(p)

File: code/57267.pas
var s, p:string;
    n,i,j:integer;
function ktdoixung(s:string):boolean;
var i,n:integer;
begin
    n:=length(s);
    i:=1;
    while (i<n) and (s[i]=s[n]) do begin
        inc(i);
        dec(j);
    end;
    exit(i>=n);
end;
BEGIN
    readln(s);
    n:=length(s);
    for i:= 1 to n do
        for j:= i to n do begin
            p:=copy(s,i,j-i+1);
            if ktdoixung(p) then writeln(p);
        end;
END.